compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -1071,7 +1071,7 @@ impl<'a> Parser<'a> {
             && self.look_ahead(1, |t| t.is_keyword(kw::Const))
             && self.look_ahead(2, |t| *t == token::CloseBracket)
```

Comment:
(I'm pretty sure we don't need the careful lookahead logic and could just `eat("[") expect("const") expect("]")` (there shouldn't be ambiguity issues (I'll post another, tangentially related PR later that goes into more details)) but that will likely regress diagnostics (because if we fail to parse a bound, we try to reparse it as a type to give `"expected trait, found type"`) and I don't want to deal with that, maybe that's why oli chose this approach